packages/datatrak-web-server/src/routes/SubmitSurveyRoute.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import { Request } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+import { DatatrakWebSubmitSurveyRequest as RequestT } from '@tupaia/types';
+import { createUpsertEntityObjects } from '../utils/createUpsertEntityObjects';
+
+export type SubmitSurveyRequest = Request<
+  RequestT.Params,
+  RequestT.ResBody,
+  RequestT.ReqBody,
+  RequestT.ReqQuery
+>;
+
+export class SubmitSurveyRoute extends Route<SubmitSurveyRequest> {
+  public async buildResponse() {
+    const { central: centralApi } = this.req.ctx.services;
+    const surveyResponseData = this.req.body;
+    const { answers, entities_upserted, country_id } = surveyResponseData;
```

Comment:
Can you possibly check that `options_created` still gets handled for the autocomplete questions?

Also the other thing is, we do a lot of processing of the response on the FE, so potentially we could move that into here if you wanted to keep this endpoint, and do the converting of the formData object into the expected shape. Then we don't have half the processing in the FE and half in the BE